tasks/kernel_matrix_testing/stacks.py

```diff
@@ -241,13 +253,24 @@ def destroy_ec2_instances(ctx, stack):
     return
 
 
+def remove_pool_directory(ctx, stack):
+    pools_dir = os.path.join(get_kmt_os().libvirt_dir, "pools")
+    for _, dirs, _ in os.walk(pools_dir):
+        for d in dirs:
+            if resource_in_stack(stack, d):
+                rm_path = os.path.join(pools_dir, d)
+                ctx.run(f"sudo rm -r {rm_path}", hide=True)
```

Comment:
Just in case, I'd add single around `{rm_path}` to avoid any possible issues with spaces and shell escaping.

```suggestion
                ctx.run(f"sudo rm -r '{rm_path}'", hide=True)
```